chia/full_node/tx_processing_queue.py

```diff
@@ -104,10 +104,22 @@ async def put(self, tx: TransactionQueueEntry, peer_id: bytes32 | None, high_pri
             self._high_priority_queue.put(tx)
         else:
             if peer_id not in self._queue_dict:
-                self._queue_dict[peer_id] = SimpleQueue()
+                self._queue_dict[peer_id] = asyncio.PriorityQueue()
                 self._index_to_peer_map.append(peer_id)
             if self._queue_dict[peer_id].qsize() < self.peer_size_limit:
-                self._queue_dict[peer_id].put(tx)
+                tx_info = tx.peers_with_tx.get(peer_id)
+                if tx_info is not None and tx_info.advertised_cost > 0:
+                    fpc = tx_info.advertised_fee / tx_info.advertised_cost
+                    priority = -fpc
+                else:
+                    # Compute the fee per cost using values from the peer with
+                    # the highest advertised cost.
+                    priority = float("inf")
+                    tx_info = max(tx.peers_with_tx.values(), key=lambda p: p.advertised_cost, default=None)
+                    if tx_info is not None and tx_info.advertised_cost > 0:
+                        fpc = tx_info.advertised_fee / tx_info.advertised_cost
+                        priority = -fpc
+                await self._queue_dict[peer_id].put((priority, tx))
```

Comment:
This else-block seems suspicious. Don't we always know the cost and fee of a transaction? At least we should know what the peer advertized it as, even if it's wrong.